Modules/Sources/WordPressMedia/ImagePrefetcher.swift

```diff
@@ -0,0 +1,108 @@
+import UIKit
+import Collections
+
+@ImageDownloaderActor
+public final class ImagePrefetcher {
+    private let downloader: ImageDownloader
+    private let maxConcurrentTasks: Int
+    private var queue = OrderedDictionary<PrefetchKey, PrefetchTask>()
```

Comment:
I feel like `OperationQueue` might be a better suit here. It manages two key things here: concurrency and queue.

Reply:
> concurrency

This part we don't really need because the entire systems already runs in the background. 

If you need an async `Operation`, I think these have outlived their usefulness in the Swift Concurrency world. There needs to be a new primitive in the standard library. But it's relatively easy to limit a maximum number of tasks even without one.

Reply:
For me personally, I probably would choose `OperationQueue` over implementing task orchestration (which is the core implementation here) myself, even though it does not feel like a "modern tech". 😆

Reply:
We should find a good open-source implementation that's sound and covered with tests. If you have something in mind, please let me know. I also recently took a stab at it https://github.com/kean/Nuke/blob/task-queue/Sources/Nuke/Internal/WorkQueue.swift, but it's not finished yet, and I'm not sure if there is a way to make it general-purpose by allowing you to customize it with an actor.